src/Methods/InvoicePaymentMethod.php

```diff
@@ -345,4 +472,89 @@ public function canHandleSubscriptions():bool
     {
         return true;
     }
+
+    /**
+     * @return bool
+     * @throws \Plenty\Exceptions\ValidationException
+     */
+    private function hasActiveShippingCountry()
+    {
+        if (empty($this->settings->getShippingCountries()) || !in_array($this->checkout->getShippingCountryId(), $this->settings->getShippingCountries())) {
+            return false;
+        } else {
+            return true;
+        }
+
+    }
+
+    /**
+     * @param int $customerId
+     * @return bool
+     */
+    private function isGuest($customerId)
+    {
+        return !$this->accountService->getIsAccountLoggedIn() && (int)$customerId === 0;
+    }
+
+    /**
+     * @param Contact $contact
+     * @return bool
+     */
+    private function isExplicitlyAllowedForThisCustomer(Contact $contact)
+    {
+        if(!$this->isGuest($contact->id)) {
+
+            if (!is_null($contact) && $contact instanceof Contact) {
+
+                $allowed = $contact->allowedMethodsOfPayment->first(function ($method) {
+                    if ($method instanceof ContactAllowedMethodOfPayment) {
+                        if ($method->methodOfPaymentId == $this->invoiceHelper->getInvoiceMopId() && $method->allowed) {
+                            return true;
+                        }
+                    }
+                });
+
+                if ($allowed) {
+                    return true;
+                } else {
+                    return false;
+                }
+            } else {
+                return false;
+
+            }
+
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * @param $lang
+     * @return bool
+     * @throws \Plenty\Exceptions\ValidationException
+     */
+    private function doNotAllowInvoiceForGuests($lang)
+    {
+        return (int)$this->settings->getSetting('disallowInvoiceForGuest',$lang) === 1;
```

Comment:
The setting 'disallowInvoiceForGuest' supports no language and as the method `SettingsService.getSetting()` has a default value "de" for the language, you do not need a `$lang` parameter.

Reply:
Are you 100% sure that it will work as expected when the shop selected language is en?
The old code is using the lang parameter:
/**
         * Check whether the user is logged in
         */
        if( $this->settings->getSetting('disallowInvoiceForGuest',$lang) == 1 && !$this->accountService->getIsAccountLoggedIn())
        {
            return false;
        }

Also, the lang parameter is used in the loadClientSettings() function and even in a query.